engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/TableTypeConversions.scala

```diff
@@ -14,19 +16,39 @@ object TableTypeConversions {
       )
     )
 
-  // not private for test
-  private[utils] def nuTypeToFlinkTableType(nuType: TypingResult): Option[DataType] = nuType match {
-    case typing.TypedObjectWithValue(underlying, _) => classToFlinkTableType(underlying.klass)
+  @tailrec
+  private def nuTypeToFlinkTableType(nuType: TypingResult): Option[DataType] = nuType match {
+    case typing.TypedObjectWithValue(typedClass, _) => nuTypeToFlinkTableType(typedClass)
     case typing.TypedClass(klass, _)                => classToFlinkTableType(klass)
-    case typing.TypedNull                           => Some(DataTypes.NULL())
+    case typing.TypedNull                           => Some(DataTypes.NULL)
     case _                                          => None
   }
 
-  // TODO: add remaining conversions
+  // As of Flink 1.19, Flink SQL does not currently support TIMESTAMP WITH TIME ZONE which should get mapped to
+  // java.time.OffsetDateTime even though it appears in Flink docs and source code.
+  // See https://issues.apache.org/jira/browse/FLINK-20869
   private def classToFlinkTableType(klass: Class[_]): Option[DataType] = klass match {
-    case klass if klass == classOf[String]                           => Some(DataTypes.STRING())
-    case klass if klass == classOf[Int] || klass == classOf[Integer] => Some(DataTypes.INT())
-    case _                                                           => None
+    case klass if klass == classOf[String] => Some(DataTypes.STRING)
+
+    case klass if klass == classOf[Boolean] || klass == classOf[java.lang.Boolean] => Some(DataTypes.BOOLEAN)
+
+    case klass if klass == classOf[Byte] || klass == classOf[java.lang.Byte]   => Some(DataTypes.TINYINT)
+    case klass if klass == classOf[Short] || klass == classOf[java.lang.Short] => Some(DataTypes.SMALLINT)
+    case klass if klass == classOf[Int] || klass == classOf[java.lang.Integer] => Some(DataTypes.INT)
+    case klass if klass == classOf[Long] || klass == classOf[java.lang.Long]   => Some(DataTypes.BIGINT)
+
+    case klass if klass == classOf[Float] || klass == classOf[java.lang.Float]   => Some(DataTypes.FLOAT)
+    case klass if klass == classOf[Double] || klass == classOf[java.lang.Double] => Some(DataTypes.DOUBLE)
+    case klass if klass == classOf[java.math.BigDecimal] => Some(DecimalTypeWithDefaultPrecisionAndScale)
+
+    case klass if klass == classOf[java.time.LocalDate]     => Some(DataTypes.DATE)
+    case klass if klass == classOf[java.time.LocalTime]     => Some(DataTypes.TIME)
+    case klass if klass == classOf[java.time.LocalDateTime] => Some(DataTypes.TIMESTAMP)
+    case klass if klass == classOf[java.time.Instant]       => Some(DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
+
+    case _ => None
   }
 
+  private val DecimalTypeWithDefaultPrecisionAndScale = DataTypes.DECIMAL(38, 8)
```

Comment:
1. Let's write an explanation, that 38 is max and 8 is arbitrary chosen reasonable scale 
2. Can you show in test this case when null is produced?

Reply:
done :+1: I wrote a test for when null was getting produced when I was testing manually but it behaves normally - it rounds the value to the default scale. I'll try to reproduce this tomorrow again

Reply:
Couldn't reproduce it now - I'm guessing it was a problem with test data or bad configuration of tables